### ISSUE TITLE
investigation: accessibility violations reported by axe

### ISSUE THREAD

**Issue by unekinn** [digdir/designsystem#2452]
We should go through these stories and determine why we get accessibility violations.

Find the story in Storybook, and open the Accessibility tab to see each violation in context

![image](https://github.com/user-attachments/assets/53b61e9c-c63c-49e4-8020-e61382656efc)

## Reported accessibility violations in these files / stories

Checked tasks have been fixed in #2400 

- [x] packages/react/stories/showcase.stories.tsx  
      story: Showcase
      violation: [Select element must have an accessible name](https://dequeuniversity.com/rules/axe/4.8/select-name?application=axeAPI)
- [x] packages/react/src/components/form/Combobox/Combobox.stories.tsx  
      story: Controlled
      violation: [Form elements must have labels](https://dequeuniversity.com/rules/axe/4.8/label?application=axeAPI)
- [x] packages/react/src/components/form/Combobox/Combobox.stories.tsx  
      story: ThousandsOfOptions
      violation: [Form elements must have labels](https://dequeuniversity.com/rules/axe/4.8/label?application=axeAPI)
- [x] packages/react/src/components/form/Combobox/Combobox.stories.tsx  
      story: RemoveAllOptions
      violation: [Form elements must have labels](https://dequeuniversity.com/rules/axe/4.8/label?application=axeAPI)
- [x] packages/react/src/components/form/Combobox/Combobox.stories.tsx  
      story: WithNumberValues
      violation: [Form elements must have labels](https://dequeuniversity.com/rules/axe/4.8/label?application=axeAPI)
- [x] packages/react/src/components/Table/Table.stories.tsx  
      story: StickyHeader
      violation: [Scrollable region must have keyboard access](https://dequeuniversity.com/rules/axe/4.8/scrollable-region-focusable?application=axeAPI)
- [x] packages/react/src/components/Table/Table.stories.tsx  
      story: WithFormElements
      violation: [Table header text should not be empty](https://dequeuniversity.com/rules/axe/4.8/empty-table-header?application=axeAPI)
- [x] packages/react/src/components/Table/Table.stories.tsx  
      story: WithFormElements
      violation: [Form elements must have labels](https://dequeuniversity.com/rules/axe/4.8/label?application=axeAPI)
- [x] packages/react/src/components/Button/Button.stories.tsx  
      story: AsLink
      violation: [`<svg>` elements with an img role must have an alternative text](https://dequeuniversity.com/rules/axe/4.8/svg-img-alt?application=axeAPI)
- [x] packages/react/src/components/Tooltip/Tooltip.stories.tsx
      story: DefaultOpen
      violation: [Elements must meet minimum color contrast ratio thresholds](https://dequeuniversity.com/rules/axe/4.8/color-contrast?application=axeAPI)
- [x] packages/react/src/components/Button/Button.stories.tsx  
      story: Icons
      violation: [Buttons must have discernible text](https://dequeuniversity.com/rules/axe/4.8/button-name?application=axeAPI)
- [x] packages/react/src/components/Button/Button.stories.tsx  
      story: IconOnly
      violation: [`<svg>` elements with an img role must have an alternative text](https://dequeuniversity.com/rules/axe/4.8/svg-img-alt?application=axeAPI)
- [x] packages/react/src/components/Button/Button.stories.tsx  
       story: IconsOnlyPrimary
       violation: [Buttons must have discernible text](https://dequeuniversity.com/rules/axe/4.8/button-name?application=axeAPI)
- [x] packages/react/src/components/Button/Button.stories.tsx  
       story: IconsOnlyPrimary
       violation: [`<svg>` elements with an img role must have an alternative text](https://dequeuniversity.com/rules/axe/4.8/svg-img-alt?application=axeAPI)
- [x] packages/react/src/components/Badge/Badge.stories.tsx  
      story: Floating
      violation: [`<svg>` elements with an img role must have an alternative text](https://dequeuniversity.com/rules/axe/4.8/svg-img-alt?application=axeAPI)
- [x] packages/react/src/components/Badge/Badge.stories.tsx  
       story: CustomPlacement
       violation: [`<svg>` elements with an img role must have an alternative text](https://dequeuniversity.com/rules/axe/4.8/svg-img-alt?application=axeAPI)
- [x] packages/react/src/components/Badge/Badge.stories.tsx  
      story: Status
      violation: [`<svg>` elements with an img role must have an alternative text](https://dequeuniversity.com/rules/axe/4.8/svg-img-alt?application=axeAPI)
- [x] packages/react/src/components/Badge/Badge.stories.tsx  
      story: InTabs
      violation: [`<svg>` elements with an img role must have an alternative text](https://dequeuniversity.com/rules/axe/4.8/svg-img-alt?application=axeAPI)
- [x] packages/react/src/components/Badge/Badge.stories.tsx  
      story: InButton
      violation: [`<svg>` elements with an img role must have an alternative text](https://dequeuniversity.com/rules/axe/4.8/svg-img-alt?application=axeAPI)
- [x] packages/react/src/components/Badge/Badge.stories.tsx  
      story: InButton
      violation: [Buttons must have discernible text](https://dequeuniversity.com/rules/axe/4.8/button-name?application=axeAPI)
- [x] packages/react/src/components/Alert/Alert.stories.tsx  
      story: MedAria
      violation: [ARIA role should be appropriate for the element](https://dequeuniversity.com/rules/axe/4.8/aria-allowed-role?application=axeAPI)
- [x] packages/react/src/components/ToggleGroup/ToggleGroup.stories.tsx  
      story: Kontrollert
      violation: [`<svg>` elements with an img role must have an alternative text](https://dequeuniversity.com/rules/axe/4.8/svg-img-alt?application=axeAPI)
- [x] packages/react/src/components/Tabs/Tabs.stories.tsx  
      story: Controlled
      violation: [`<svg>` elements with an img role must have an alternative text](https://dequeuniversity.com/rules/axe/4.8/svg-img-alt?application=axeAPI)
- [x] packages/react/src/components/loaders/Skeleton/Skeleton.stories.tsx  
      story: TextExample
      violation: [Elements must meet minimum color contrast ratio thresholds](https://dequeuniversity.com/rules/axe/4.8/color-contrast?application=axeAPI)
- [x] packages/react/src/components/form/Switch/Switch.stories.tsx  
      story: FullWidth
      violation: [Elements must meet minimum color contrast ratio thresholds](https://dequeuniversity.com/rules/axe/4.8/color-contrast?application=axeAPI)
- [x] packages/react/src/components/form/Switch/Switch.stories.tsx  
      story: FullWidthRight
      violation: [Elements must meet minimum color contrast ratio thresholds](https://dequeuniversity.com/rules/axe/4.8/color-contrast?application=axeAPI)

## Violations we will ignore

These violations are examples of invalid usage of components, but these stories only exist to test sizing/alignment, so we don't want to do the work to add correct aria attributes.

- packages/react/stories/testing.stories.tsx  
   story: MediumRow
- packages/react/stories/testing.stories.tsx  
   story: SmallRow
- packages/react/stories/testing.stories.tsx  
   story: LargeRow
- packages/react/stories/testing.stories.tsx  
   story: SmallCol
- packages/react/stories/testing.stories.tsx  
   story: MediumCol
- packages/react/stories/testing.stories.tsx  
   story: LargeCol
